source/protocol/lsp-messages.adb

```diff
@@ -1121,6 +1121,58 @@ package body LSP.Messages is
       V : ALS_ShowDependenciesParams)
    renames LSP.Message_IO.Write_ALS_ShowDependenciesParams;
 
+   procedure Write_CreateFile
+     (S : access Ada.Streams.Root_Stream_Type'Class;
+      V : LSP.Messages.CreateFile)
+   is
+      JS : LSP.JSON_Streams.JSON_Stream'Class renames
+        LSP.JSON_Streams.JSON_Stream'Class (S.all);
+      Kind : constant LSP.Types.LSP_String := LSP.Types.To_LSP_String
+        (Ada.Strings.UTF_Encoding.UTF_8_String'("create"));
+   begin
+      JS.Start_Object;
+      JS.Key ("kind");
+      LSP.Types.Write (S, Kind);
```

Comment:
Perhaps we can add 'kind' fields to CreateFile/DeleteFile/RenameFile record types and use auto-generated Read/Write subprograms.
```ada
kind : Document_Change_Kind range Create_File .. Create_File := Create_File;
```

Not sure if we shall use `Document_Change_Kind` or another dedicated enumeration type. I can do this if you want.

Reply:
Agree, that would definitely allow us to auto generate the Read/Write subprograms for these types, so I think we should do that. I'll work on that and update this pull request.